packages/mobile/src/app/WebViewScreen.tsx

```diff
@@ -25,22 +26,31 @@ type RouteProps = StackScreenProps<StackParamList, Screens.WebViewScreen>
 type Props = RouteProps
 
 function WebViewScreen({ route, navigation }: Props) {
-  const { uri, headerTitle } = route.params
+  const { headerTitle, uri, dappkitDeeplink } = route.params
+
   const dispatch = useDispatch()
   const { t } = useTranslation()
+  const dappSessionActive = useSelector(dappSessionActiveSelector)
 
-  const webviewRef = useRef<WebViewRef>(null)
+  const webViewRef = useRef<WebViewRef>(null)
   const [canGoBack, setCanGoBack] = useState(false)
   const [canGoForward, setCanGoForward] = useState(false)
 
+  const handleCloseWebView = () => {
+    if (dappSessionActive) {
+      dispatch(dappSessionEnded())
```

Comment:
This needs to be called when the webView is unmounted.
Which not only happens when the Close button is touched, but also when swiping back on iOS or when the Android back button is used.

We could use a `useEffect` clean up function for this.

Reply:
ah yes good one, i hadn't thought of this! i have added this now